app/views/shops/_form.html.haml

```diff
@@ -25,4 +25,23 @@
 			= f.label :email_id
 			= f.email_field :email_id
 			%br/
-			= f.submit(class: "btn btn-primary")
\ No newline at end of file
+			Location:
+			%input#address{:style => "width: 200px", :type => "text"}/
+			#map{:style => "width: 500px; height: 400px;"}
+			= f.hidden_field :latitude
+			= f.hidden_field :longitude
+			:javascript
```

Comment:
```
Location:
%input#address{:style => "width: 200px", :type => "text"}/
#map{:style => "width: 500px; height: 400px;"}
= f.hidden_field :latitude
= f.hidden_field :longitude
:javascript
	var lat = parseFloat($('#shop_latitude').val())
	var lon = parseFloat($('#shop_longitude').val())
	$('#map').locationpicker({
		location: {latitude: lat || 53.348864, longitude: lon || -6.2502467},
		radius: 0,
		enableAutocomplete: true,
		inputBinding: {
			latitudeInput: $('#shop_latitude'),
			longitudeInput: $('#shop_longitude'),
			locationNameInput: $('#address')
		}
	});
%br/
= f.submit(class: "btn btn-primary")
```

Check the code above, the reason I didn't save it as an external file is the rails `id`, such as `:latitude`, `:latitude` is differed based on the view. I need to hard code them in the JavaScript, that's why in JS part the `id` are `#shop_latitude` and `#shop_longitude`.